### DISCUSSION TITLE
How to get indexName and params from <InstantSearch />?

### DISCUSSION THREAD

**Discussion by robikovacs** [algolia/instantsearch#5308] [Q&A] [answered]
Hi everyone,

I'm trying to implement a "Select All" feature and I will need the current filter params and index name of the InstantSearch component? Is there a way to do this? [Here is an example](https://jsfiddle.net/s3ap8pew/1/) of a widget done with instantsearch.js

Refs:
* https://discourse.algolia.com/t/how-to-get-all-hits-using-instantsearch-js/3115
* https://jsfiddle.net/s3ap8pew/1/

Best,
Robert

**Comment by dhayab** [comment, marked as answer]
Hi @robikovacs , with React InstantSearch, you can use the connector [`connectStateResults`](https://www.algolia.com/doc/api-reference/widgets/state-results/react/) to retrieve your search state as well as the index name.

Here is a sandbox showing it in action: https://codesandbox.io/s/agitated-darwin-7iwk5y?file=/src/App.js.

**Reply by robikovacs** [reply]
That's great! Thank you @dhayab 🙌

**Comment by robikovacs** [comment]
@dhayab One issue that I found with this approach is that the initial state is not correct if there is `defaultRefinement` set for example.

Here is an example: https://codesandbox.io/s/elegant-beaver-oztkz6

There it would be expected to see something like this on page load:
```json
{
  "index": "demo_ecommerce",
  "state": {
    "refinementList": {
      "brand": [
        "Incipio"
      ]
    },
    "page": 1,
    "configure": {
      "hitsPerPage": 8
    }
  }
}
```

**Comment by gloireOndongoDy** [comment]
In the app I am building, users can opt to select and download all documents associated with the current search. When the search returns only a 10 or less results, this is easy because all the info I need is available in this hits. However, when the results span multiple pages, I don’t have access to the data I need to facilitate a “Download All.”

Was wondering then, can I programmatically get all hits so that I can get the information I need to implement this? That is, I need the get an array of Obj.file_name for each object in the search’s complete result set.
